aten/src/ATen/native/TensorShape.cpp

```diff
@@ -3216,7 +3216,19 @@ Tensor view(const Tensor& self,
 }
 
 Tensor alias(const Tensor& self) {
-    return alias_with_sizes_and_strides(self, self.sizes(), self.strides());
+  if (self.is_sparse_csr()) {
```

Comment:
Thanks for adding this! You can expand this to more sparse formats by using `self.is_sparse_csc`, `self.is_sparse_bsr`, `self.is_sparse_bsc`. Can you also update our unit tests for alias to try these layouts? You can construct sparse tensors via `to_sparse_csr` etc. from 2D dense Tensors.

Reply:
test_transpose in test_sprase_csr.py is already testing with all_sparse_compressed_layouts.
Does it imply other sparse format is supporting regular .alias() call ? 

```
    @all_sparse_compressed_layouts()
    def test_transpose(self, device, dtype, layout):
```

Reply:
@SherlockNoMad - I thought we have standalone tests for alias that need to be extended with support for these additional layouts.

Reply:
After rebasing to the latest master, I found that a recent PR already handled sparse tensors with sparse_compressed_transpose function. 
So this special case for alias is no longer needed. 

I am removing this now.

Reply:
Handled in this PR https://github.com/pytorch/pytorch/pull/82122